Simulator/Core/Driver.cpp

```diff
@@ -56,38 +49,88 @@
+///
+///  @param  argc      argument count.
+///  @param  argv      arguments.
+///  @returns    true if successful, false otherwise.
+bool parseCommandLine(string cmdLineArguments)
+{
+   ParamContainer cl;       // todo: note as third party class.
+   cl.initOptions(false);   // don't allow unknown parameters
+   cl.setHelpString(string(
+      "The UW Bothell graph-based simulation environment, for high-performance neural network and other graph-based problems."));
+   // ("\nUsage: "+ argv[0] + " ");
+
+   // Set up the comment line parser.
+   if ((cl.addParam("configfile", 'c', ParamContainer::filename, "parameter configuration filepath")
+        != ParamContainer::errOk)
+#if defined(USE_GPU)
+       || (cl.addParam("device", 'd', ParamContainer::regular, "CUDA device id")
+           != ParamContainer::errOk)
+#endif   // USE_GPU
+       || (cl.addParam("stimulusfile", 's', ParamContainer::filename, "stimulus input filepath")
+           != ParamContainer::errOk)
+       || (cl.addParam("deserializefile", 'r', ParamContainer::filename,
+                       "simulation deserialization filepath (enables deserialization)")
+           != ParamContainer::errOk)
+       || (cl.addParam("serializefile", 'w', ParamContainer::filename,
+                       "simulation serialization filepath (enables serialization)")
+           != ParamContainer::errOk)
+       || (cl.addParam("version", 'v', ParamContainer::novalue,
+                       "output current git commit ID and exit")
+           != ParamContainer::errOk)) {
+      cerr << "Internal error creating command line parser" << endl;
+      return false;
+   }
+
+   // Parse the command line
+   if (cl.parseCommandLine(cmdLineArguments) != ParamContainer::errOk) {
+      cl.dumpHelp(stderr, true, 78);
+      return false;
+   }
+
+   if (cl["version"].compare("") != 0) {
+      cout << "Git commit ID: " << GIT_COMMIT_ID << endl;
+      exit(0);
+   }
 
-///  Main for Simulator. Handles command line arguments and loads parameters
+   // Get the command line values
+   Simulator::getInstance().setConfigFileName(cl["configfile"]);
+   Simulator::getInstance().setDeserializationFileName(cl["deserializefile"]);
+   Simulator::getInstance().setSerializationFileName(cl["serializefile"]);
+   Simulator::getInstance().setStimulusFileName(cl["stimulusfile"]);
+
+
+#if defined(USE_GPU)
+   if (EOF == sscanf(cl["device"].c_str(), "%d", &g_deviceId)) {
+      g_deviceId = 0;
+   }
+#endif   // USE_GPU
+   return true;
+}
+
+///  setupSimulation handles command line arguments and loads parameters
 ///  from parameter file. All initial loading before running simulator in Network
-///  is here.
+///  is done here.
 ///
 ///  @param  argc    argument count.
 ///  @param  argv    arguments.
-///  @return -1 if error, else if success.
-int main(int argc, char *argv[])
+///  @return -1 if error, else 0 if success.
+int Driver::setupSimulation(string cmdLineArguments)
```

Comment:
Let's call this `runSimulation()`.

Reply:
Modified the method name from `setupSimulation `to `runSimulation`